.jenkins/Jenkinsfile

```diff
@@ -11,762 +11,534 @@
+      sh "mkdir -p build/stage-logs"
+    } else {
+      unstash name: "${arch}_${jdk}"
+    }
+}
+
+def getNodeLabel(command, cell) {
+  echo "using node label: cassandra-${cell.arch}-${command.size}"
+  return "cassandra-${cell.arch}-${command.size}"
+}
+
+def copyToNightlies(sourceFiles, remoteDirectory='') {
+    if (!isCanonical() || "" == sourceFiles) {
+      return;
+    }
+
+    def remotePath = remoteDirectory.startsWith("cassandra/") ? "${remoteDirectory}" : "cassandra/${JOB_NAME}/${BUILD_NUMBER}/${remoteDirectory}"
+    def attempt = 1
+    retry(9) {
+      if (attempt > 1) { sleep(60 * attempt) }
+      sshPublisher(
+      continueOnError: true, failOnError: false,
+      publishers: [
+        sshPublisherDesc(
+        configName: "Nightlies",
+        transfers: [ sshTransfer( sourceFiles: sourceFiles, remoteDirectory: remotePath) ]
+        )
+      ])
+    }
+    echo "archived to https://nightlies.apache.org/${remotePath}"
+}
+
+def cleanAgent(job_name) {
+  if (isCanonical()) {
+    def maxJobHours = 12
+    echo "Cleaning project, and pruning docker for '${job_name}' on ${NODE_NAME}…" ;
+    sh """
+        git clean -qxdff -e build/test/jmh-result.json || true;
+        if pgrep -xa docker || pgrep -af "build/docker" || pgrep -af "cassandra-builds/build-scripts" ; then docker system prune --all --force --filter "until=${maxJobHours}h" || true ; else  docker system prune --force --volumes || true ;  fi;
+      """
+  }
+}
+
+//  CASSANDRA-18130
+def saveAgentReport() {
+  if (isCanonical()) {
+    //
+    // echo "Updating disk usage report…";
+    // sh """
+    //     ( echo "----" ;
+    //     echo \$(date) ;
+    //     echo "${JOB_NAME} ${BUILD_NUMBER} ${STAGE_NAME}" ;
+    //     du -xm / 2>/dev/null | sort -rn | head -n 30 ;
+    //     df -h ) | tee -a \$(date +"%Y%m%d%H%M")-disk-usage-stats.txt
+    //   """
+    //   copyToNightlies("*-disk-usage-stats.txt", "cassandra/agents/${NODE_NAME}/disk-usage/")
+    //   sh 'rm *-disk-usage-stats.txt'
+  }
+}
+
+/////////////////////////////////////////
+////// scripting support for summary ////
+/////////////////////////////////////////
+
+def generateTestReports() {
+  node('cassandra-medium||cassandra-amd64-medium') {
+    cleanWs()
+    checkout changelog: false, scm: scmGit(branches: [[name: params.branch]], extensions: [cloneOption(depth: 1, noTags: true, reference: '', shallow: true)], userRemoteConfigs: [[url: params.repository]])
+    copyArtifacts filter: 'test/**/TEST-*.xml.xz,test/**/cqlshlib*.xml.xz,test/**/nosetests*.xml.xz', fingerprintArtifacts: true, projectName: env.JOB_NAME, selector: specific(env.BUILD_NUMBER), target: "build/", optional: true
+    if (fileExists('build/test/output')) {
+      // merge splits for each target's test report, other axes are kept separate
+      //   TODO parallelised for loop
+      //   TODO results_details.tar.xz needs to include all logs for failed tests
+      sh """
```

Comment:
We can inline this script into the `generate-ci-summary.sh` as it is used only for Jenkins.

Reply:
Not really.  There's structuring of files this is dependent on that only happens in this Jenkinsfile.

I do believe there's opportunity to push this along with the path structuring of test result files down.  But this has to be done in consultation with other CI systems (that changes in build.xml and the scripts would impact).

Also note the challenge here with docker-in-docker…